src/GraphQL.Tests/Bugs/Issue2387_OverrideBuiltInScalars.Alt.cs

```diff
@@ -0,0 +1,259 @@
+using System.Threading.Tasks;
+using GraphQL.Language.AST;
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Bugs
+{
+    // https://github.com/graphql-dotnet/graphql-dotnet/issues/2387
+    public class Issue2387_OverrideBuiltInScalars_Alt : QueryTestBase<Issue2387_OverrideBuiltInScalars_Alt.MySchema>
+    {
+        public Issue2387_OverrideBuiltInScalars_Alt()
+        {
+            Services.Singleton<MySchema>();
```

Comment:
I was worth me some time to understand why the tests failed, and the replacement of the scalars did not work. It turned out that by default schema works as a transient dependency and a new one is created every time accessing `QueryTestBase.Schema`.